src/main/java/org/galatea/starter/domain/HistoricalPricesCache.java

```diff
@@ -0,0 +1,117 @@
+package org.galatea.starter.domain;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+
+import java.io.*;
+import java.time.LocalDate;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Slf4j
+public class HistoricalPricesCache {
+    private String cacheFilePath;
+
+    private Map<String, Map<String, List<IexHistoricalPrice>>> rangedCache;
+    private Map<String, Map<LocalDate, List<IexHistoricalPrice>>> datedCache;
+
+    public HistoricalPricesCache(String cacheFilePath) {
+        this.cacheFilePath = cacheFilePath;
+
+        log.info("Attempting to load historical prices cache from disk. Loading from {}", cacheFilePath);
+
+        try(FileInputStream fi = new FileInputStream(new File(cacheFilePath))) {
+            log.info("Cache file found. Initialising cache...");
+
+            try (ObjectInputStream oi = new ObjectInputStream(fi)) {
+                rangedCache = (Map<String, Map<String, List<IexHistoricalPrice>>>) oi.readObject();
+                datedCache = (Map<String, Map<LocalDate, List<IexHistoricalPrice>>>) oi.readObject();
+
+                log.info("Cache loaded. Loaded {} ranged symbols and {} dated symbols", rangedCache.size(), datedCache.size());
+            }
+            catch (ClassNotFoundException ex) {
+                log.warn("Exception raised when loading from cache:\n{}\nInitialising empty cache.", ex);
+                rangedCache = new HashMap<>();
+                datedCache = new HashMap<>();
+            }
+
+        }
+        catch (FileNotFoundException ex) {
+            log.info("No cache file at {} found. Initialising empty cache.", cacheFilePath);
+            rangedCache = new HashMap<>();
+            datedCache = new HashMap<>();
+        }
+        catch (IOException ex) {
+            log.warn("IOException when attempting to load cache file {}\n{}", cacheFilePath, ex);
+            log.warn("Cache could not be loaded. Initialising empty cache.", cacheFilePath);
```

Comment:
Either the log message is missing a placeholder for cacheFilePath, or cacheFilePath doesn't need to be included. Presumably the latter, since cacheFilePath is already mentioned in the previous log